apps/swirl-docs/src/components/Navigation/SidebarNavigation/index.tsx

```diff
@@ -0,0 +1,54 @@
+import { isProdDeployment } from "@swirl/lib/env";
+import { NavItem, navItems } from "@swirl/lib/navigation/";
+import classNames from "classnames";
+import { useRouter } from "next/router";
+import { SidebarNavSubItem } from "./SidebarNavSubItem";
+
+export const ANIMATION_OPENED = { opacity: 1, height: "auto" };
+export const ANIMATION_CLOSED = { opacity: 0, height: 0 };
+
+export function SidebarNavigation() {
+  const router = useRouter();
+  const activePath = router.asPath;
+
+  const pathBase = activePath.split("/")[1];
+  const navItemBase = navItems.find((item) =>
+    item.url?.startsWith("/" + pathBase)
+  );
+
+  if (navItemBase?.devOnly && isProdDeployment) {
+    return null;
+  }
+
+  const navChildren = navItemBase?.children;
+
+  if (!navChildren?.length) {
+    return null;
```

Comment:
please use undefined. null is an object in javascript and not a "real" null value.

Reply:
<img width="121" alt="image" src="https://github.com/getflip/swirl/assets/23700195/722b3e18-127f-4c90-b415-34f0142753f3">
vs 

<img width="166" alt="image" src="https://github.com/getflip/swirl/assets/23700195/54f15e12-e286-46b3-9985-ad319075f9ba">

Reply:
I don't get this to work with TypeScript. 
```
'SidebarNavigation' cannot be used as a JSX component.
  Its return type 'Element | null | undefined' is not a valid JSX element.
    Type 'undefined' is not assignable to type 'Element | null'.ts(2786)
```

Reply:
alright. then we'll go with null